app/models/project.rb

```diff
@@ -137,22 +137,22 @@ def to_hash(recurse = true)
       path: UrlGenerator.new.project_path(self),
       hidden: hidden,
       featured: featured,
-      likeCount: likes.count,
+      likeCount: likes.length,
       content: content,
       timeAgoInWords: time_ago_in_words(created_at),
       createdAt: created_at.strftime('%B %d, %Y'),
       ownerName: owner.nil? ? nil : owner.name,
       ownerUrl: owner.nil? ? nil : UrlGenerator.new.user_url(owner),
-      dataSetCount: data_sets.count,
-      dataSetIDs: data_sets.pluck(:id),
-      fieldCount: fields.count,
+      dataSetCount: data_sets.length,
+      dataSetIDs: data_sets.to_a.map(&:id),
```

Comment:
You might be right, it is hard to know with out profiling it.  New Relic is a great way get a sense of this.  But just for sake of argument :).  Let look at some real numbers. This project on isense: https://isenseproject.org/projects/864 has 2,228 datasets and 201 media objects. 

So first the visualization will create 2,228 dataset objects. Then it will create all of those dataset objects again for each media object (when the media object runs to_hash on its project).  So that will be 450,056 dataset objects being created.  If we say each dataset takes 100 bytes (it is probably more than that), that will around 45 MB that isn't too bad, but if it is 1k per object (more likely) it will be 450MB. That is quite of bit of memory for one process to handle and when multiple requests come in at the same time it could be a problem.  If we look at time.